umbra-js/src/classes/Umbra.ts

```diff
@@ -380,19 +380,35 @@ export class Umbra {
     const startBlock = overrides.startBlock || this.chainConfig.startBlock;
     const endBlock = overrides.endBlock || 'latest';
 
+    const filterSupportedAddresses = async (announcements: AnnouncementDetail[]): Promise<AnnouncementDetail[]> => {
+      const filtered = await Promise.all(
+        announcements.map(async (i) => {
+          const [isReceiverSupported, isFromSupported] = await Promise.all([
+            assertSupportedAddress(i.receiver),
+            assertSupportedAddress(i.from),
+          ]);
+          return isReceiverSupported && isFromSupported ? i : null;
+        })
+      );
+
+      return filtered.filter((i) => i !== null) as AnnouncementDetail[];
```

Comment:
Hmm a regular for loop feels a bit grosser since you have to collect all the promises and push them within the loop, so I'm ok with the type safety tradeoff since it's straightforward here